src/queue.c

```diff
@@ -0,0 +1,101 @@
+#include "queue.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#define TRUE 1
+#define FALSE 0
+
+struct node
```

Comment:
A think a better name for `node` would be [packet](https://en.wikipedia.org/wiki/Network_packet). `struct packet_t`.